mantidproject/mantid
# Project Recovery: Handle algorithms that require a string list of workspace names

Currently High priority because of a bug present in master with regards to project recovery that popped up recently and will be present in the nightly build if someone were to try recovering grouped workspaces mantid would tell them it is broken however this would not be the truth.

**Background of work**
GroupWorkspaces was originally removed from being recovered and it WAS working fine however due to a change recently in the way it checks whether workspaces are back which I suspect is due to a internal Qt change but could also be a master change, I am working on fixing the problem instead of the symptom.

Currently project recovery stores all history of workspaces in the files, this includes Saving to files, so the idea is to stop it from calling these saved files, the best way to leave the other algorithms still functioning normally was to stop it from saving specific files.

**Description of work.**
Removed the previous way of handling GroupWorkspaces as it will no longer after required pull requests be required as it is handled properly now. I also stopped the saving of save based algorithms so now it will not re-write files to disk on recovery

**To test:**
- Group two workspaces
- Wait for project recovery to save
- Run [`Segfault(DryRun=False)`](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/Segfault.cpp)
- Load up mantid
- Attempt recovery
- The group should be there with it's respective workspaces

and this test aswell

 - Load another workspace (From a nexus file would be easier)
 - Run SaveNexus on this workspace
 - Wait for Project Recovery to save (using the debug level of logging in the logger)
- Run [`Segfault(DryRun=False)`](https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/src/Segfault.cpp)
- Reload mantid
 - Click open script in scripting window it should only have the Load algorithm calls, and not the  SaveNexus.


Fixes #23452 and #23450

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
